File: python_code/Map_Codes/b3_IDW_plots.py
import numpy as np
import matplotlib.pyplot as plt
import os


def cart_map(mypolyLat, mypolyLon, mod_Lat, mod_Lon, conc, explat, explon):
    list_vars = ['Cl', 'SO4', 'PO4', 'U', 'CaH', 'MgH', 'HCO3']
    name_labels = 'abcdefg'

    s = plt.figure(figsize=(12, 10))
    s.set_facecolor('white')

    for j in range(7):
        ax = s.add_subplot(3, 3, j + 1)
        var_name = list_vars[j]

        if isinstance(conc, dict) and var_name in conc:
            Z = conc[var_name]
        else:
            Z = np.full(mod_Lat.shape, np.nan)

        pc = ax.pcolormesh(mod_Lon, mod_Lat, Z, cmap='jet', shading='flat')
        plt.colorbar(pc, ax=ax, fraction=0.046, pad=0.04)

        if len(mypolyLat) > 2:
            ax.plot(mypolyLon, mypolyLat, 'k-', linewidth=1)

        ax.scatter(explon, explat, s=10, c='black', marker='o', zorder=5)

        ax.set_xlabel('Longitude')
        ax.set_ylabel('Latitude')
        ax.set_title(f'({name_labels[j]})')
        ax.grid(True, alpha=0.3)

    plt.tight_layout()
    return s


def b3_IDW_plots():
    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    save_dir = os.path.join(base_dir, 'saved_models')

    try:
        data = np.load(os.path.join(save_dir, 'geoplott_2.npz'), allow_pickle=True)
        jprLat = data['jprLat']
        jprLon = data['jprLon']
        mod_Lat = data['mod_Lat']
        mod_Lon = data['mod_Lon']
        mod_conc = data['mod_conc'].item()
        explat = data['explat']
        explon = data['explon']
    except FileNotFoundError:
        data = np.load(os.path.join(save_dir, 'geoplott_3.npz'), allow_pickle=True)
        jprLat = data['jprLat']
        jprLon = data['jprLon']
        mod_Lat = data['mod_Lat']
        mod_Lon = data['mod_Lon']
        mod_conc = data['mod_conc'].item()
        explat = data['explat']
        explon = data['explon']

    fig = cart_map(jprLat, jprLon, mod_Lat, mod_Lon, mod_conc, explat, explon)
    return fig

File: Map Codes/b3_IDW_plots.m
function b3_IDW_plots()
close all
clc
clearvars
%%%%%%%%%%%%%%%%%%%%%%%%%

load geoplott_3.mat jprLat jprLon mod_Lat mod_Lon mod_conc explat explon
%load geoplott.mat jprLat jprLon mod_Lat mod_Lon mod_conc explat explon %F, NO3
%load geoplott_2.mat jprLat jprLon mod_Lat mod_Lon mod_conc explat explon %GQ,LQ,AQ,RQ



cart_map(jprLat,jprLon,mod_Lat,mod_Lon,mod_conc,explat,explon)

disp('')

function cart_map(mypolyLat,mypolyLon,mod_Lat,mod_Lon,conc,explat,explon)
%%Plot the interpolated surface within the shape limits
list={'Cl','SO4','PO4','U','CaH','MgH','HCO3'};
%list={'F','NO3'};
%list={'GQ','LQ','AQ','RQ'};
name='a':'z';
s=figure ();
set(gcf,'color','w');
for j=1:7
Z=conc.(list{j});
subplot(3,3,j)
hold on;
hi=pcolor(mod_Lon,mod_Lat,Z);
b=hi.Parent.YTickLabel;c=strcat(b,'^{\circ}N');hi.Parent.YTickLabel=c;
b=hi.Parent.XTickLabel;c=strcat(b,'^{\circ}E');hi.Parent.XTickLabel=c;
shading flat; %interp;
colormap(jet);
colorbar;
%%%%%%%%%%%%%%%%%%%
pos=hi.Parent.Position;
a = pos(1); b = pos(2); c = pos(3); d = pos(4);
xl=hi.Parent.XLim; yl=hi.Parent.YLim;
p1 = a + (86.3 - xl(1)) / (xl(2) - xl(1)) * c;
q1 = b + (21 - yl(1)) / (yl(2) - yl(1)) * d;
p2 = a + (86.3 - xl(1)) / (xl(2) - xl(1)) * c;
q2 = b + (21.1 - yl(1)) / (yl(2) - yl(1)) * d;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
annotation('textarrow',[p1 p2],[q1 q2],'String','N')
%northarrow('position',[86.7 21.1 1.5 1.5]);
grid on
plot(mypolyLon,mypolyLat,'k'); % Plot the shape boundaries
scatter(explon,explat,10,'filled',"black")
% labels=1:length(explat);
% cellstr(num2str(labels));
% text(explon+0.01,explat+0.01,cellstr(num2str(labels')),'HorizontalAlignment','left')
xlabel('Longitude');
ylabel('Latitude');
title(['(' name(j) ')']);
box on 
grid on
hold off
end
copygraphics(s,'Resolution',600,'BackgroundColor','white','Padding','tight','PreserveAspectRatio','on') 
disp('');





